test/services/candlepin/message_handler_test.rb

```diff
@@ -136,5 +136,13 @@ def test_delete_pool
       @handler.delete_pool
       assert_empty Katello::Pool.find_by(:cp_id => @pool_id)
     end
+
+    def test_delete_pool_on_null
+      pool = @handler.pool
+      @handler.stubs(:pool).returns(pool).then.returns(nil)
```

Comment:
Could we just delete the pool instead of stub?

Reply:
deleting the pool wont fail it ? The double stub is to simulate the original race condition.

Reply:
Fair, but this stub also does nothing now :)